chunk/table_manager.go

```diff
@@ -85,37 +123,35 @@ type PeriodicTableConfig struct {
 
 // RegisterFlags adds the flags required to config this to the given FlagSet
 func (cfg *PeriodicTableConfig) RegisterFlags(f *flag.FlagSet) {
-	f.BoolVar(&cfg.UsePeriodicTables, "dynamodb.use-periodic-tables", true, "Should we user periodic tables.")
+	f.BoolVar(&cfg.UsePeriodicTables, "dynamodb.use-periodic-tables", true, "Should we use periodic tables.")
 	f.StringVar(&cfg.TablePrefix, "dynamodb.periodic-table.prefix", "cortex_", "DynamoDB table prefix for the periodic tables.")
 	f.DurationVar(&cfg.TablePeriod, "dynamodb.periodic-table.period", 7*24*time.Hour, "DynamoDB periodic tables period.")
 	f.Var(&cfg.PeriodicTableStartAt, "dynamodb.periodic-table.start", "DynamoDB periodic tables start time.")
 }
 
 // DynamoTableManager creates and manages the provisioned throughput on DynamoDB tables
 type DynamoTableManager struct {
-	dynamoDB  StorageClient
-	tableName string
-	cfg       TableManagerConfig
-	done      chan struct{}
-	wait      sync.WaitGroup
+	dynamoDB DynamoTableClient
+	cfg      TableManagerConfig
+	done     chan struct{}
+	wait     sync.WaitGroup
 }
 
 // NewDynamoTableManager makes a new DynamoTableManager
-func NewDynamoTableManager(cfg TableManagerConfig) (*DynamoTableManager, error) {
-	dynamoDBClient, tableName := cfg.mockDynamoDB, cfg.mockTableName
-	if dynamoDBClient == nil {
-		var err error
-		dynamoDBClient, tableName, err = NewDynamoDBClient(cfg.DynamoDB.String())
-		if err != nil {
-			return nil, err
+func NewDynamoTableManager(cfg TableManagerConfig, dynamoDBClient DynamoTableClient, originalTableName string) (*DynamoTableManager, error) {
+	if originalTableName != "" {
```

Comment:
Please put a note in the tech debt column on the TODO project so we don't forget to remove all this.